src/dataflow-types/src/plan/reduce.rs

```diff
@@ -576,3 +914,22 @@ fn reduction_type(func: &AggregateFunc) -> ReductionType {
         | AggregateFunc::LagLead { .. } => ReductionType::Basic,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use mz_repr::proto::protobuf_roundtrip;
+    use proptest::prelude::*;
+
+    // This test causes stack overflows if not run with --release,
+    // ignore by default.
```

Comment:
Do we know which struct/enum is causing the stack overflow? Can we implement `Arbitrary` for that struct/enum?

Reply:
@aalexandrov  already ran into this issue and started looking into it. The best solution would be to improve proptest to not put everything on the stack